invenio_oauthclient/contrib/cern.py

```diff
@@ -212,13 +212,34 @@ def account_setup(remote, token):
 
     if response.status == requests.codes.ok:
         res = get_dict_from_response(response)
-        current_user.info['group'] = fetch_groups(res['Group'])
+        groups = fetch_groups(res['Group'])
+        current_user.info['group'] = groups
         current_user.modified = True
         current_user.save()
 
-        if user and not any([user.family_name, user.given_names]):
-            user.family_name = res['Lastname'][0]
-            user.given_names = res['Firstname'][0]
-
-            db.session.add(user)
-            current_user.reload()
+        # update user object
+        if user:
```

Comment:
I'm thinking that the cern client (as well as the other clients) should only be a thin layer that handles specific client data, and all the logic and making sure that the user/token/etc. are well created and made available to the application should be taken care of in the oauthclient handlers - which I think is already the case. account_setup function should not even be called if anything is wrong with creating/fetching the user.